### ISSUE TITLE
Possibility to return native python UUID type

### ISSUE THREAD

**Issue by PrinceJunkie** [aminalaee/uuid-utils#32]
Would it be possible to add a parameter, that a native python UUID is returned instead of the uuid-utils UUID type?

Background: Some packages like psycopg2 expect a native python UUID returned back instead of sth. customized.

**Comment by aminalaee** [comment]
I can take a look into this, can you share a traceback about what is happening? How the typechecking is done in psycopg.

**Comment by PrinceJunkie** [comment]
Thanks for your great work and looking into this!

Here's some example code to reproduce the issue:
```python3
import uuid

import psycopg2
import psycopg2.extras
import uuid_utils

psycopg2.extras.register_uuid()

# This would work
# generated_uuid = uuid.uuid4()

# This generated_uuid will fail
generated_uuid = uuid_utils.uuid4()

conn = psycopg2.connect(dbname="postgres", user="postgres", password="postgres", host="localhost")
cursor = conn.cursor()

sql_query = "INSERT INTO test (id) VALUES (%s);"
cursor.execute(sql_query, [generated_uuid])

conn.commit()

```

The given output is:
```
Traceback (most recent call last):
  File "/dev/uuid_utils_test.py", line 19, in <module>
    cursor.execute(sql_query, [generated_uuid])
psycopg2.ProgrammingError: can't adapt type 'uuid_utils.UUID'
```

The function to enable UUID acceptance (`psycopg2.extras.register_uuid()`) can be found at: https://github.com/psycopg/psycopg2/blob/ad5bee7054519d87f25bc5828c502b2ebe197049/lib/extras.py#L647

**Comment by aminalaee** [comment]
I think it can be done with some tricks but the problem is it won't be that fast, for example:

```py
import uuid
import uuid_utils

def custom_uuid4():
    return uuid.UUID(int=uuid_utils.uuid4().int)
```

But benchmarking this it will be ~2-3 times faster than Python, instead of ~10 times.

**Comment by PrinceJunkie** [comment]
Yes, I know that this will be significantly slower than the native implementation but such an integrated method could be useful for others too.

**Comment by lepenkinya** [comment]
Looks like the problem can be solved by registering an additional adapter:

```
from psycopg2.extensions import register_adapter, ISQLQuote
from psycopg2.extras import UUID_adapter
import uuid_utils

register_adapter(uuid_utils.UUID, UUID_adapter)
```

**Comment by PrinceJunkie** [comment]
Could you tell me how you can do this with sqlalchemy?

But regardless of my specific issue with psycopg2/sqlalchemy there are other tools like pydantic which would need the native type.

**Comment by opentyler** [comment]
I would also be very interested in this feature, definitely a hurdle many people will come across. Your suggested trick (uuid.UUID(int=uuid_utils.uuid4().int)) works well enough even tho it's slower. Thanks again for the great library!
